packages/common/src/hooks/useListViewData/useSortBy.ts

```diff
@@ -0,0 +1,27 @@
+import { useCallback, useState } from 'react';
+
+export interface SortRule<T> {
+  key: keyof T;
+  isDesc: boolean;
+}
+
+interface SortState<T> {
+  sortBy: { key: keyof T; direction: 'asc' | 'desc'; isDesc: boolean };
+  onChangeSortBy: (newSortKey: keyof T) => void;
+}
+
+export const useSortBy = <T>(initialSortBy: SortRule<T>): SortState<T> => {
+  const [{ key, isDesc }, setSortBy] = useState(initialSortBy);
+
+  const onChangeSortBy = useCallback((newSortKey: keyof T) => {
+    setSortBy(({ key: prevSortKey, isDesc: prevIsDesc }) => {
+      const newIsDesc = prevSortKey === newSortKey ? !prevIsDesc : false;
+      return { key: newSortKey, isDesc: newIsDesc };
+    });
+  }, []);
+
+  return {
+    sortBy: { key, isDesc, direction: isDesc ? 'desc' : 'asc' },
+    onChangeSortBy,
+  };
```

Comment:
passing `direction` is because some mui stuff wants `'desc'` or `'asc'` so just did it here.. why not 🤷 

I change this hook a little later to accept just a `keyof T` as the `initialSortBy`. This was silly and I will revert it back to using this implementation of passing a `SortRule`. Also might need to change the `onChangeSortBy` to accept a `SortRule`? Though, it is nice to just be able to pass a string, so .. maybe both?